taccsite_cms/middleware/cms_template.py

```diff
@@ -0,0 +1,49 @@
+"""Change CMS page template at runtime"""
+import os
+
+from cms.middleware.toolbar import ToolbarMiddleware
+from cms.models.pagemodel import Page as CMS_Page
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+
+class CMSTemplateMiddleware(ToolbarMiddleware):
```

Comment:
Just checking: is this the most descriptive name for the middleware class name?

Reply:
I don't know. What I know:

- It allows one to explicitly swap to one of the [`CMS_TEMPLATES`](https://docs.django-cms.org/en/3.11.8/reference/configuration.html#cms-templates).
- It supports shorthand for implicit swap i.e. `?raw`.
- It seems typical to prepend the name of a Django class with the kind of class it is.

What about these names:

- `CMSTemplateSwapMiddleware`
- `CMSTemplateChangeMiddleware`
- `CMSTemplateOverrideMiddleware`

Reply:
Any of those proposed is great, thanks @wesleyboar

Reply:
Renamed… but **PR closed**, because the `?template=….html` feature [already works by default](https://github.com/TACC/Core-CMS/pull/872#issuecomment-2641341128). 😅